### PR TITLE
IGNITE-11282 fix primitive arrays usage as cache index

### PR DESCRIPTION
**Pull request by iakkuratov**

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java [hunk 1]
```diff
@@ -78,8 +54,17 @@
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
 
+import javax.cache.Cache;
+import javax.cache.processor.EntryProcessor;
+import javax.cache.processor.EntryProcessorException;
+import javax.cache.processor.MutableEntry;
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.util.*;
```
Comment (zstan) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java]: forbidden asterisk, replace with direct imports.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/KeyCacheObjectImpl.java [hunk 2]
```diff
@@ -198,7 +217,7 @@ public KeyCacheObjectImpl(Object val, byte[] valBytes, int part) {
 
         KeyCacheObjectImpl other = (KeyCacheObjectImpl)obj;
 
-        return val.equals(other.val);
```
Comment (ivandasch) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/KeyCacheObjectImpl.java]: Equality of hashcodes is not enough for equality of objects or arrays.
Please,  use Arrays.equals here

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java [hunk 3]
```diff
@@ -1200,6 +1201,70 @@ public void testTransform() throws Exception {
...
+                equals(new KeyCacheObjectImpl(new float[]{1,2,3},null,1)) : "Not equals";
+
+        assert new KeyCacheObjectImpl(testIdx[4],null,1).
+                equals(new KeyCacheObjectImpl(new double[]{1,2,3},null,1)) : "Not equals";
+
+        assert new KeyCacheObjectImpl(testIdx[5],null,1).
+                equals(new KeyCacheObjectImpl(new long[]{1,2,3},null,1)) : "Not equals";
+
+        assert new KeyCacheObjectImpl(testIdx[6],null,1).
+                equals(new KeyCacheObjectImpl(new char[]{'a','b','c'},null,1)) : "Not equals";
```
Comment (zstan) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java]: 4 spaces left here

File: modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java [hunk 4]
```diff
@@ -1200,6 +1201,70 @@ public void testTransform() throws Exception {
         }
     }
 
+    /**
+     * IGNITE-11282
```
Comment (zstan) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/GridCacheBinaryObjectsAbstractSelfTest.java]: replace plz

**Comment by iakkuratov** [comment]
Fixed by IGNITE-12116

**Pull request closed without merging**